app/src/test/java/net/squanchy/service/firestore/model/schedule/FirestoreTracksRepositoryTest.kt

```diff
@@ -0,0 +1,70 @@
+package net.squanchy.service.firestore.model.schedule
+
+import io.reactivex.Observable
+import io.reactivex.observers.TestObserver
+import net.squanchy.schedule.domain.view.Track
+import net.squanchy.schedule.domain.view.aTrack
+import net.squanchy.service.firestore.FirestoreDbService
+import net.squanchy.service.firestore.aFirestoreTrack
+import net.squanchy.service.repository.firestore.FirestoreTracksRepository
+import net.squanchy.support.lang.Optional
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.junit.MockitoJUnit
+import org.mockito.junit.MockitoRule
+
+class FirestoreTracksRepositoryTest {
+
+    @Rule
+    @JvmField
+    var rule: MockitoRule = MockitoJUnit.rule()
+
+    @Mock
+    private lateinit var firestoreDbService: FirestoreDbService
+
+    @Test
+    fun `should emit an empty list of tracks when there are no tracks on the DB`() {
+        val trackService = FirestoreTracksRepository(firestoreDbService)
+        val subscription = TestObserver<List<Track>>()
+        `when`(firestoreDbService.tracks()).thenReturn(Observable.just(emptyList()))
+
+        trackService.tracks()
+            .subscribe(subscription)
```

Comment:
just checking, I think you can do `.test().assertValue(emptyList())` instead of having to create the subscription and stuff

Reply:
Yeah I think so, I'm just used to doing it this way. Can improve later on?

Reply:
sure thing